great_expectations/core/usage_statistics/package_dependencies.py

```diff
@@ -0,0 +1,229 @@
+"""Provide GE package dependencies.
+
+This module contains static lists of GE dependencies, along with a utility for
+checking and updating these static lists.
+
+    Typical usage example:
+        ge_dependencies: GEDependencies = GEDependencies()
+        print(ge_dependencies.get_required_dependency_names())
+        print(ge_dependencies.get_dev_dependency_names())
+
+    To verify lists are accurate, you can run this file or execute main() from
+    within a cloned GE repository. This will check the existing requirements
+    files against the static lists returned via the methods above in the
+    usage example and raise exceptions if there are discrepancies.
+"""
+import os
+import re
+from typing import List, Set
+
+
+class GEDependencies:
+    """Store and provide dependencies when requested.
+
+    Also acts as a utility to check stored dependencies match our
+    library requirements.
+
+    Attributes: None
+    """
+
+    """This list should be kept in sync with our requirements.txt file."""
+    GE_REQUIRED_DEPENDENCIES: List[str] = [
+        "altair",
+        "Click",
+        "colorama",
+        "cryptography",
+        "dataclasses",
+        "importlib-metadata",
+        "Ipython",
+        "jinja2",
+        "jsonpatch",
+        "jsonschema",
+        "mistune",
+        "nbformat",
+        "numpy",
+        "packaging",
+        "pandas",
+        "pyparsing",
+        "python-dateutil",
+        "pytz",
+        "requests",
+        "ruamel.yaml",
+        "scipy",
+        "termcolor",
+        "tqdm",
+        "typing-extensions",
+        "urllib3",
+        "tzlocal",
+    ]
+
+    """This list should be kept in sync with our requirements-dev*.txt files."""
+    GE_DEV_DEPENDENCIES: List[str] = [
```

Comment:
How do we determine what is important to track? Could an argument be made to gather as much information as possible and put the burden of parsing/munging on analytics?

I know we have concerns about PII but what if we parsed all our requirements files and then only provided information about those libraries? That would allow us a lot of flexibility, even if the requirements rapidly changed.

Reply:
With all the data, we can always backtrack and say "oh hey that library is actually more important to track than we thought"

Reply:
I'm now realizing this is an executable script and you're actually verifying against the actual files - I'm still going to keep these comments here but feel free to resolve if they aren't applicable.

Reply:
I don't think we need to track dev dependencies; certainly not wholesale. I would be interested in tracking packages that are used to integrate in specific environments though. Ken did some work to break these out, maybe they would be useful here.

Reply:
Great points @cdkini . We don't want to track the full environment since we only want to know what version of a GE dependency is installed so that we can adjust our requirements bounds, and we don't want to inadvertently track a non-public package for example that pierces anonymization. And correct - we are verifying against the files in CI and via this utility so hopefully it is easy and enforced to keep these lists in sync. Very open to changing this if there is a better way that does not require manual update but I have not yet found one.

Reply:
Very good point @donaldheppner. Those dependencies for specific environments are captured here since they fit the `requirements-dev-*.txt` naming pattern e.g. `requirements-dev-mysql.txt`. If we want to separate dev vs specific environment requirements via a different naming pattern it would be simple to update this script. cc @kenwade4 
Also we are pretty permissive with our dev requirements, it may be good to track these as well so that we can update the bounds to more recent versions.